d2l-scroll-wrapper.html

```diff
@@ -0,0 +1,88 @@
+<link rel="import" href="../polymer/polymer.html">
+<link rel="import" href="../iron-resizable-behavior/iron-resizable-behavior.html">
+
+<!--
+# D2L Scroll Wrapper
+
+## Usage
+```
+	<link rel="import" href="d2l-scroll-wrapper.html">
+	<style>
+		#scroll-wrapper[h-scrollbar] {
+			/* styles to apply when horizontal scrollbar appears */
+		}
+		#scroll-wrapper[scrollbar-left] {
+			/* styles to apply when scrolled to the left */
+		}
+		#scroll-wrapper[scrollbar-right] {
+			/* styles to apply when scrolled to the right */
+		}
+	</style>
+	<d2l-scroll-wrapper id="scroll-wrapper">
+		<div>Stuff</div>
+	</d2l-scroll-wrapper>
+```
+-->
+<dom-module id="d2l-scroll-wrapper">
+	<template strip-whitespace>
+		<style>
+			:host {
+				display: block;
+				overflow: auto;
+				width: 100%;
+				box-sizing: border-box;
+			}
+		</style>
+		<content></content>
+	</template>
+	<script>
+		Polymer({
+			is: 'd2l-scroll-wrapper',
+			behaviors: [Polymer.IronResizableBehavior],
+
+			properties: {
+				/** Whether this element has a horizontal scrollbar */
+				hScrollbar: {
+					type: Boolean,
+					reflectToAttribute: true,
+					value: false,
+					readOnly: true
+				},
+				/** Whether this element is scrolled to the left */
+				scrollbarLeft: {
+					type: Boolean,
+					reflectToAttribute: true,
+					value: false,
+					readOnly: true
+				},
+				/** Whether this element is scrolled to the right */
+				scrollbarRight: {
+					type: Boolean,
+					reflectToAttribute: true,
+					value: false,
+					readOnly: true
+				}
+			},
+
+			listeners: {
+				'iron-resize': 'checkScrollbar',
+				'scroll': 'checkScrollThresholds'
+			},
+
+			attached: function() {
+				this.async(this.checkScrollbar);
+			},
+
+			checkScrollbar: function() {
+				this._setHScrollbar(this.offsetWidth !== this.scrollWidth);
+				this.checkScrollThresholds();
+			},
+
+			checkScrollThresholds: function() {
+				var lowerScrollValue = this.scrollWidth - this.offsetWidth - this.scrollLeft;
+				this._setScrollbarLeft(this.scrollLeft === 0);
+				this._setScrollbarRight(lowerScrollValue <= 0);
```

Comment:
I think I'm missing something... where is `_setScrollbarLeft` defined? Is it a special Polymer thing for setting readonly properties?